### ISSUE TITLE
Inferred output is NoReturn when it shouldn't

### ISSUE THREAD

**Issue by EricSalemi** [microsoft/pyright#3645]
## Environment data

    Visual Studio: 1.68.1
    Pylance v2022.6.30
    CPython 3.10.4
    MacOS 12.4
    
## Expected behaviour

Inferred method output should be `Any`.

## Actual behaviour

Inferred method output is`NoReturn`.

## Code Snippet

```python
class Helper:
    def return_if_one_value(self, iterable):
        count = None
        for count, value in zip(range(1), iterable):
            pass
        match count:
            case None:
                raise Exception('Too few values!')
            case 0:
                return value
            case _:
                raise Exception('Too many values!')
```

**Comment by erictraut** [comment]
Moving to pyright since this is a core type checker issue.

**Comment by EricSalemi** [comment]
> initializing value to None

I worked around it by adding an output argument type hint but I prefer your workaround.

Thanks for the quick reaction!

**Comment by EricSalemi** [comment]
@erictraut Not sure why but initializing `value` to `None` still results in inferring `NoReturn` for me.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.257, which I just published. It will also be included in the next release of pylance.
